snap7/types.py

```diff
@@ -254,6 +255,41 @@ class S7CpuInfo(ctypes.Structure):
         ('ModuleName', ctypes.c_char * 25)
     ]
 
+    @property
```

Comment:
thank you for your contribution! But I'm not sure if this is a great idea, since it shadows the local variables which are already dynamically created. It works, but it is error-prone and might be confusing. what is the problem you are trying to solve? Are some of these fields optional for the PLC you are using?

Reply:
Yeah, it shadows the attribute tags, it was on purpose. It was about your enhancement ticket about pythonic "classes". If we want to use the attribute you need to go to source and search for the attribute names, here there are built-in in the python object. Moreover for **S7CpuInfo**, you can direct convert the byte string values in str to facilitate the handling.

I can convert it to classic get function, if shadowing the C-struct is the problem. And add some more documentation on it ?

Reply:
the problem is that this is not backwards compatible, and it might become an issue if someone wants to access the raw values without conversion. You assume all users want a string in a specific encoding, which I think might not be the case.

I'm tempted to just leave it as is and leave the decoding to the end user, unless we can come up with an elegant way to convert his automatically.